sos/plugins/juju.py

```diff
@@ -40,82 +22,142 @@ class Juju(Plugin, UbuntuPlugin):
     files = ('/usr/bin/juju', '/usr/bin/juju-run', '/snap/bin/juju')
 
     option_list = [
-        ('export-mongodb',
-         'Export mongodb collections as json files', '', False),
-        ('generate-bundle',
-         """Generate a YAML bundle of the current environment
-         (requires juju-deployerizer)""", '', False),
+        ("juju-user", "The user to run juju commands as", "", "ubuntu")
     ]
 
-    def get_deployed_services(self):
-        cmd = "juju status --format json"
-        status_json = self.call_ext_prog(cmd)['output']
-        self.add_string_as_file(status_json, "juju_status_json")
-        # if status_json isn't in JSON format (i.e. 'juju' command not found),
-        # or if it does not contain 'services' key, return empty list
+    # Get the configuration of each application deployed in a model
+    def collect_app_config(self, username, modelname, statusfilespath):
+        statusinfo = {}
+        try:
+            fp = open(statusfilespath, "r")
+            statusinfo = json_loads(fp.read())
+            fp.close()
+        except Exception:
+            return
+        if "applications" in statusinfo:
+            for appname, appinfo in statusinfo["applications"].items():
+                # Get the application config for every application in every
+                # model
+                self.add_cmd_output(
+                    "sudo -i -u {} juju config --format yaml -m {} {}".format(
+                        username, modelname, appname
+                    )
+                )
+
+    # Get some information pertaining to each model
+    def collect_model_output(self, username, modelsfilepath):
+        modelsinfo = {}
         try:
-            return json_loads(status_json)['services'].keys()
-        except ValueError:
-            return []
-
-    @ensure_service_is_running("juju-db")
-    def export_mongodb(self):
-        collections = (
-            "relations",
-            "environments",
-            "linkednetworks",
-            "system",
-            "settings",
+            fp = open(modelsfilepath, "r")
+            modelsinfo = json_loads(fp.read())
```

Comment:
switch it to with 

open(modelsfilepath, 'r') as fd: 
  modelsinfo = jsn_loads(fd.read())

no close is required